ext/standard/string.c

```diff
@@ -5687,18 +5687,18 @@ PHP_FUNCTION(str_pad)
 	}
 
 	if (pad_str_len == 0) {
-		php_error_docref(NULL, E_WARNING, "Padding string cannot be empty");
+		zend_throw_error(NULL, "Padding string cannot be empty");
 		return;
 	}
 
 	if (pad_type_val < STR_PAD_LEFT || pad_type_val > STR_PAD_BOTH) {
-		php_error_docref(NULL, E_WARNING, "Padding type has to be STR_PAD_LEFT, STR_PAD_RIGHT, or STR_PAD_BOTH");
+		zend_throw_error(NULL, "Padding type has to be STR_PAD_LEFT, STR_PAD_RIGHT, or STR_PAD_BOTH");
 		return;
 	}
 
 	num_pad_chars = pad_length - ZSTR_LEN(input);
 	if (num_pad_chars >= INT_MAX) {
-		php_error_docref(NULL, E_WARNING, "Padding length is too long");
+		zend_throw_error(NULL, "Padding length is too long");
```

Comment:
This error condition looks bogus: I see no reason why the number of padding characters should be limited by INT_MAX. Could you submit a separate PR to drop this entirely?

Reply:
Sure, maybe something due to before PHP could handle large strings ? (I remember reading something about 2GB string size limit)